mlir/lib/Conversion/AIRRtToIpuPass.cpp

```diff
@@ -344,6 +344,34 @@ static LogicalResult CastFunctionArgs(func::FuncOp funcOp,
     rewriter.setInsertionPointToStart(&entry);
     auto cast = rewriter.create<UnrealizedConversionCastOp>(
         rewriter.getUnknownLoc(), memrefTy, entry.getArgument(i));
+    // With memref shape collapsed to 1d, the multi-dimensional offset also
+    // needs to be collapsed.
+    SmallVector<Operation *> users;
+    for (auto user : entry.getArgument(i + 1).getUsers()) {
+      if (auto cast = dyn_cast<UnrealizedConversionCastOp>(user)) {
```

Comment:
`cast` shadows the variable from line 345. best to pick another name.

Reply:
Done.

Reply:
Variable `i` is also shadowing another `i` too.

Reply:
Thanks for spotting this, @newling ! Done.